test/math_test.cpp

```diff
@@ -0,0 +1,41 @@
+#include "src/math.hpp"
+
+#include "boost/ut.hpp"
+
+#include <cmath>
+
+// NOLINTBEGIN(readability-magic-numbers)
+
+auto main() -> int
+{
+    using namespace boost::ut;
+    using opt::dual;
+
+    constexpr dual x{1.0F, 2.0F};
+
+    test("dualnumbers math exp") = [&x] {
+        expect(eq(std::exp(1.0F), opt::exp(1.0F)));
```

Comment:
What about a compile-time check against a magic number? Or should we pull in `gcem` and test against that (or use it)?

Reply:
Yes. Today if I have time I plan to finish the hyperdual numbers stuff and then I can pull gcem